src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1571,12 +1571,17 @@ void RequireDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionC
 				} else if (uniqueValue is SystemTypeValue systemTypeValue) {
 					MarkTypeForDynamicallyAccessedMembers (ref reflectionContext, systemTypeValue.TypeRepresented, requiredMemberTypes);
 				} else if (uniqueValue is KnownStringValue knownStringValue) {
-					TypeDefinition foundType = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents)?.Resolve ();
+					TypeReference typeRef = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents);
+					TypeDefinition foundType = typeRef?.Resolve ();
 					if (foundType == null) {
 						// Intentionally ignore - it's not wrong for code to call Type.GetType on non-existing name, the code might expect null/exception back.
 						reflectionContext.RecordHandledPattern ();
 					} else {
 						MarkTypeForDynamicallyAccessedMembers (ref reflectionContext, foundType, requiredMemberTypes);
+						if (typeRef is GenericInstanceType genericType) {
+							foreach (var arg in genericType.GenericArguments)
+								MarkType (ref reflectionContext, arg);
+						}
```

Comment:
This is not necessary and in also not complete. The issue at hand is about generics, but the same problem applies to `T[]` and/or `T*` and so on. All of this is already handled by `MarkType` - which calls `GetOriginalType` which does the recursive walk of the element types and/or generic arguments.

So all we need to do here is call `MarkType(typeRef)`. It should not include "more" as marking a type really only guarantees presence of the type, but none of its members.